code/modules/mob/living/carbon/human/human_life.dm

```diff
@@ -903,42 +868,248 @@
 			to_chat(H, "<span class='warning'>You smell something foul...</span>")
 			H.fakevomit()
 
+/mob/living/carbon/human/proc/handle_blood_pressure()
+	var/datum/organ/heart/H = get_int_organ_datum(ORGAN_DATUM_HEART)
+	var/datum/organ/lungs/L = get_int_organ_datum(ORGAN_DATUM_LUNGS)
+	if(!H || !L || stat == DEAD || (NO_BLOOD in dna.species.species_traits)) // No heart, no heartbeat
+		blood_pressure = 0
+		return
+
+	var/chem_volume = reagents.total_volume
+	var/temp_bp = (blood_volume * 0.016) ** 2 // Baseline. At 560u of blood you have a bp of 80
+	if(chem_volume)
+		temp_bp += round(chem_volume * 0.25, 1)
+
+	var/obj/item/organ/internal/heart/heart = H.linked_organ
+	if(istype(heart))
+		temp_bp += heart.blood_pressure_change
+
+	if(!isdrask(src) && !isdiona(src)) // These species live longer than normal humans, exempt
+		if(age > 65) // Hey gramps, watch what you're doing
+			temp_bp += 15
+		else if(age > 45) // Getting old
+			temp_bp += 10
+
+	if(HAS_TRAIT(src, TRAIT_FAT))
+		temp_bp += 10
+
+	temp_bp += L.linked_organ.damage
+	temp_bp += 5 * get_infected_limbs()
+
+	var/found_alcohol = FALSE
+	for(var/datum/reagent/chem in reagents.reagent_list)
+		if(istype(chem, /datum/reagent/consumable/ethanol))
+			if(found_alcohol)
+				continue
+			found_alcohol = TRUE
+
+		temp_bp += chem.blood_pressure_change
+
+	temp_bp += 5 * length(reagents.addiction_list)
+
+	switch(temp_bp)
+		if(0 to BLOODPRESSURE_DANGER)
+			Dizzy(3 SECONDS)
+			adjustOxyLoss(5)
+			adjustBrainLoss(1) // Brains are more fragile, no cap (on the damage)
+			if(!H.linked_organ.is_robotic())
+				H.linked_organ.receive_damage(1, TRUE)
+
+			if(!L.linked_organ.is_robotic())
+				L.linked_organ.receive_damage(1, TRUE)
+
+			if(prob(10))
+				to_chat(src, "<span class='warning'>You feel incredibly weak.</span>")
+
+		if(BLOODPRESSURE_DANGER to BLOODPRESSURE_LOW)
+			if(prob(5))
+				Dizzy(5 SECONDS)
+			// Oxyloss under 60 BP
+			adjustOxyLoss(max(round(7 - temp_bp / 10, 1), 1))
+		if(100 to 130)
+			if(prob(5))
+				to_chat(src, "<span class='warning'>You feel a bit weak.</span>")
+
+		if(BLOODPRESSURE_NORMAL to BLOODPRESSURE_V_HIGH)
+			false_pain += rand(1, 10)
+			if(prob(5))
+				to_chat(src, "<span class='warning'>Your nose bleeds.</span>")
+				bleed(10)
+			if(prob(5))
+				SetEyeBlind(5 SECONDS)
+			else if(prob(5))
+				Dizzy(5 SECONDS)
+
+		if(BLOODPRESSURE_V_HIGH to INFINITY)
+			false_pain += rand(5, 15)
+
+			if(prob(5))
+				for(var/obj/item/organ/external/limb in shuffle(bodyparts))
+					if(limb.status & ORGAN_INT_BLEEDING)
+						continue
+					limb.cause_internal_bleeding()
```

Comment:
Maybe add some sort of distinctive message here to indicate why folks just got internal bleeding out of (seemingly) nowhere? Maybe something like
"You feel feel like something popped in [limb]"